R/partial.R

```diff
@@ -43,45 +50,67 @@
 #' f()
 #' f()
 #'
-#' # You can override this by saying .lazy = FALSE
-#' f <- partial(runif, n = rpois(1, 5), .lazy = FALSE)
+#' # You can override this by unquoting:
+#' f <- partial(runif, n = !! rpois(1, 5))
+#' f
+#' f()
+#' f()
+#'
+#' # Partialized arguments can be spliced:
+#' args <- list(n = rpois(1, 5))
+#' f <- partial(runif, !!! args)
 #' f
 #' f()
 #' f()
 #'
-#' # This also means that partial works fine with functions that do
-#' # non-standard evaluation
+#' # partial() works fine with functions that do non-standard evaluation
 #' my_long_variable <- 1:10
 #' plot2 <- partial(plot, my_long_variable)
 #' plot2()
 #' plot2(runif(10), type = "l")
-partial <- function(...f, ..., .env = parent.frame(), .lazy = TRUE,
-                    .first = TRUE) {
-  stopifnot(is.function(...f))
+#'
+partial <- function(...f, ..., .first = TRUE, .env, .lazy) {
+  if (!missing(.env))
+    abort("`.env` is deprecated")
+  if (!missing(.lazy))
+    abort("`.lazy` is deprecated; unquote values to evalute them immediately")
 
-  if (.lazy) {
-    fcall <- substitute(...f(...))
-  } else {
-    fcall <- make_call(substitute(...f), .args = list(...))
-  }
+  f <- as_closure(...f)
```

Comment:
Would `as_mapper()` be more consistent with the rest of purrr?

Reply:
For consistency, `as_mapper()` might indeed be a better choice. I hadn't considered that. I slightly prefer `as_closure()` because I think it more clearly expresses why `...f` is being transformed: we need to apply arguments by name, and that generally won't work for primitive functions.